### PR TITLE
Make generated mock classes open by default

### PR DESCRIPTION
**Pull request by mpdifran**
This addresses #170

### CODE REVIEW

**Comment by mpdifran** [comment]
This addresses #170

**Comment by amichnia** [comment]
@mpdifran thank you. I see one potential problem - if classes are public, shouldn't mean as well that init's and methods should be public? Same with Given, Perform and Verify generated classes - creation methods should be public probably as well.

**Comment by mpdifran** [comment]
So I've found the correct template to edit, and I've regenerated the mocks for the tests. The mocks are now open, but their methods are not. I'm not sure how I can specify the visibility for these. Should I be editing something in the template, or elsewhere?

**Comment by Przemyslaw-Wosko** [comment]
@mpdifran yes, but you will end up also with adding improvements for properties and computed properties

**Comment by mpdifran** [comment]
I'm not sure what you mean. Which file should I look in to edit the generated properties?

**Comment by amichnia** [comment]
@mpdifran Hey, the case is:

- We are currently in transition from old template (one big file), into "new system" (Templates folder). That means, that most of the changes should be done in Files in Templates directory. There is a script in Rakefile `rake template` that will combine them into the big template.
- That means that the approach should be:
  1. Make changes to files in Templates dir
  2. Regenerate big mock file with `rake template`

Also, making mocks open would not be enough. To be able to just import mock module (without testable) you will need to make public:
  - Given, Perform, Verify nested classes
  - all constructor methods that creates Given, Perform etc (in MethodWrapper.swift I think)
  - maybe something more, I need to check the generated mock

File: SwiftyMocky-Tests/macOS/Mocks/ItemsRepositoryMock.swift [hunk 1]
```diff
@@ -150,7 +150,7 @@ class ItemsRepositoryMock: ItemsRepository, Mock {
         static func storedDetails(item: Parameter<Item>) -> Verify { return Verify(method: .m_storedDetails__item_item(`item`))}
```
Comment (amichnia) [inline SwiftyMocky-Tests/macOS/Mocks/ItemsRepositoryMock.swift]: these static functions should also be public.

**Comment by mpdifran** [comment]
Ok, I'll update when I get a chance.

**Comment by mpdifran** [comment]
I was able to get the tests running with the open mocks. Let me know if I need to change anything!

**Comment by amichnia** [comment]
@mpdifran looking at it

**Pull request merged**